src/libraries/Common/src/Extensions/ParameterDefaultValue/ParameterDefaultValue.cs

```diff
@@ -39,21 +43,27 @@ public static bool TryGetDefaultValue(ParameterInfo parameter, out object? defau
                     defaultValue = parameter.DefaultValue;
                 }
 
+                bool isNullableParameterType = parameter.ParameterType.IsGenericType &&
+                    parameter.ParameterType.GetGenericTypeDefinition() == typeof(Nullable<>);
+
                 // Workaround for https://github.com/dotnet/runtime/issues/18599
-                if (defaultValue == null && parameter.ParameterType.IsValueType)
+                if (defaultValue == null && parameter.ParameterType.IsValueType
```

Comment:
Why didn't we need this check before?

Reply:
Previously we were calling `Activator.CreateInstance`.

```C#
Type t = typeof(Nullable<int>);
Console.WriteLine($"Activator Is Null? {Activator.CreateInstance(t) == null}");
Console.WriteLine($"GetUninitializedObject Is Null? {RuntimeHelpers.GetUninitializedObject(t) == null}");
```

prints

```
Activator Is Null? True
GetUninitializedObject Is Null? False
```

https://github.com/dotnet/runtime/pull/44843#issuecomment-729344082 says `System.Nullable<T> (will be treated as simply T)` for `GetUninitializedObject`.